tests/IBTK/cf_interface.cpp

```diff
@@ -0,0 +1,236 @@
+        // Initialize the AMR patch hierarchy.
+        gridding_algorithm->makeCoarsestLevel(patch_hierarchy, 0.0);
+        int tag_buffer = 1;
+        int level_number = 0;
+        bool done = false;
+        while (!done && (gridding_algorithm->levelCanBeRefined(level_number)))
+        {
+            gridding_algorithm->makeFinerLevel(patch_hierarchy, 0.0, 0.0, tag_buffer);
+            done = !patch_hierarchy->finerLevelExists(level_number);
+            ++level_number;
+        }
+
+        // Allocate patch data
+        int coarsest_ln = 0, finest_ln = patch_hierarchy->getFinestLevelNumber();
+        for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            level->allocatePatchData(Q_idx);
+        }
+
+        // Fill in initial data
+        for (int ln = 0; ln <= patch_hierarchy->getFinestLevelNumber(); ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+            {
+                Pointer<Patch<NDIM> > patch = level->getPatch(p());
+                Pointer<CartesianPatchGeometry<NDIM> > pgeom = patch->getPatchGeometry();
+                const double* const dx = pgeom->getDx();
+                const double* const xlow = pgeom->getXLower();
+                const hier::Index<NDIM>& idx_low = patch->getBox().lower();
+                Pointer<CellData<NDIM, double> > Q_cc_data = patch->getPatchData(Q_idx);
+                Pointer<SideData<NDIM, double> > Q_sc_data = patch->getPatchData(Q_idx);
+                if (Q_cc_data)
+                {
+                    for (CellIterator<NDIM> ci(patch->getBox()); ci; ci++)
+                    {
+                        const CellIndex<NDIM>& idx = ci();
+                        VectorNd x;
+                        for (int d = 0; d < NDIM; ++d)
+                            x[d] = xlow[d] + dx[d] * (static_cast<double>(idx(d) - idx_low(d)) + 0.5);
+                        (*Q_cc_data)(idx) = Q_fcn(x);
+                    }
+                }
+                else if (Q_sc_data)
+                {
+                    for (int axis = 0; axis < NDIM; ++axis)
+                    {
+                        for (SideIterator<NDIM> si(patch->getBox(), axis); si; si++)
+                        {
+                            const SideIndex<NDIM>& idx = si();
+                            VectorNd x;
+                            for (int d = 0; d < NDIM; ++d)
+                                x[d] = xlow[d] +
+                                       dx[d] * (static_cast<double>(idx(d) - idx_low(d)) + (d == axis ? 0.0 : 0.5));
+                            (*Q_sc_data)(idx) = Q_fcn(x);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Once #1640 gets merged, we can replace this with a `CartGridPointwiseFunction` object.